src/controller/error-controller.ts

```diff
@@ -0,0 +1,20 @@
+import type { FastifyInstance, FastifyError } from "fastify";
+
+type HttpErrorResponse = {
+  statusCode: number;
+  details?: Record<string, string>;
+} & FastifyError;
+
+export function setErrorHandler(fastify: FastifyInstance) {
```

Comment:
The Fastify framework provides this `setErrorHandler` to add custom functionality to error responses: https://www.fastify.io/docs/latest/Reference/Server/#seterrorhandler